packages/react-dom-bindings/src/client/ReactDOMInput.js

```diff
@@ -114,10 +103,13 @@ export function initWrapperState(element: Element, props: Object) {
 
   const node = ((element: any): InputWithWrapperState);
   const defaultValue = props.defaultValue == null ? '' : props.defaultValue;
-
+  const initialChecked =
+    props.checked != null ? props.checked : props.defaultChecked;
   node._wrapperState = {
     initialChecked:
-      props.checked != null ? props.checked : props.defaultChecked,
+      typeof initialChecked !== 'function' &&
+      typeof initialChecked !== 'symbol' &&
```

Comment:
There's a subtle change to the attribute table here.

Because defaultChecked was set here to the value and then that was used by its truthy value in postMountWrapper to set the defaultChecked value. That's not the semantics for boolean properties elsewhere and not what the server does.

This fixes that.